src/gol.php

```diff
@@ -61,23 +86,36 @@
     $maxIteration = intval($getOpt->getOption('m'));
 }
 
-if ($getOpt->getOption("startRandom"))
-{
-    $field = new BoardRandom($width, $height);
-}
-else if ($getOpt->getOption("startGlider"))
-{
-    $field = new BoardGlider($width, $height);
-}
-else if ($getOpt->getOption("startAcorn"))
+
+if ($getOpt->getOption("input"))
 {
-    $field = new BoardAcorn($width, $height);
+    $arg = $getOpt->getOption("input");
+
+    foreach ($inputs as $type => $input)
+    {
+        if ($type == $arg)
+        {
+            $field = new Board($width, $height);
+            $input->prepareBoard($field, $getOpt);
+            break;
+        }
+    }
 }
-else
+
+if ($field == null)

```

Comment:
What is this block used for? Since input is required, shouldn't this never happen?